### ISSUE TITLE
Android & iOS-Chat - Report flickers, keyboard is dismissed when focus on composer on Android

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45152]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.6-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Launch New Expensify app
2. Go to workspace chat
3. Submit some expenses
4. Click on the expense preview
5. Go back and forth between main chat, expense report and expense thread

## Expected Result:
App will not flicker when switching between expense reports

## Actual Result:
App flickers when switching between expense reports
On Android, keyboard is dismissed when trying to focus on the composer

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/93ee54d3-8554-45b0-a8a1-4bb47667e01c

https://github.com/Expensify/App/assets/78819774/ce4958d0-4156-426c-8afe-8a135ddc10cf

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@danieldoglas  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by danieldoglas** [comment]
I don't think this is related to backend, removing the `DeployBlocker` tag

**Comment by izarutskaya** [comment]
@danieldoglas Also the app is flickering when a new user finishes the onboarding flow. It flickers again after tapping on the "Get started" button. The issue can't be reproduced with the same account after this.
Let me know if we need to create a new issue. Thanks

https://github.com/Expensify/App/assets/115492554/8d14198a-7335-44cb-8dad-8fd6b65716f6

**Comment by Julesssss** [comment]
The [one PR](https://github.com/Expensify/App/pull/44511/files) that modified text input behaviour seems unlinked, but it would be worth testing out just to be sure.

**Comment by rushatgabhane** [comment]
> The [one PR](https://github.com/Expensify/App/pull/44511/files) that modified text input behaviour seems unlinked, but it would be worth testing out just to be sure.

tested. its unrelated

**Comment by danieldoglas** [comment]
Maybe lets's check reverting https://github.com/Expensify/App/pull/45120 and https://github.com/Expensify/App/pull/45120

@rushatgabhane @allgandalf

**Comment by rushatgabhane** [comment]
@Julesssss closed it here - https://github.com/Expensify/App/pull/45255

**Comment by danieldoglas** [comment]
@rushatgabhane I agree, but that was reverting just the patch, I'm suggesting we revert the patch and the update

**Comment by Julesssss** [comment]
@danieldoglas yeah that sounds good. Happy to help test if you can share a PR

**Comment by Julesssss** [comment]
A promising-looking fix is being discussed [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1720787675322989?thread_ts=1720776606.394749&cid=C01GTK53T8Q). 

However even with a fix Android builds will fail, and the staging build will remain bricked. Because of this I am continuing to keep my focus on the Android workflow issue [here](https://github.com/Expensify/Expensify/issues/412320).

**Comment by ikevin127** [comment]
> Bumped the price for this one to $500
> ...
> If anyone can run git bisect to identify the PR that we can revert to fix this, the reward wont miss you.

:money_mouth_face: I'm here to collect the bounty & help unblock the deploy at the same time 🙂

After some manual bisect and lots of android builds -> I got it down to this commit from last week (July 8th):
- [Merge pull request #44460 from margelo/feat/react-compiler-healthcheck](https://github.com/Expensify/App/commit/fe2fcf02e3c8120eb33a1bcd4d7720e96c333e0b)
- PR -> https://github.com/Expensify/App/pull/44460

Before ^ this commit, building the app on [this commit](https://github.com/Expensify/App/commit/a5fb7cd8304d2005246deeadb83d727d5276cb6c) (right before the offending one) -> the flicker / composer keyboard issue is not present on Android (didn't check iOS but I'm pretty sure it's caused by the same offending commit).

Cheers! :beer:

**Comment by kirillzyusko** [comment]
@ikevin127 would you mind to share a video? Basically in this PR I:
- bumped `babel-plugin-react-compiler` to the newest version and it started to compile 1000 files instead of 600;
- added healthcheck dependency and patched it (but it shouldn't affect the app, since this library is not used during the compilation);
- bumped `eslint-plugin-react-compiler` - also shouldn't affect the app, since it's used in eslint only 🤷‍♂️ 

I also tried to disable `react-compiler` (the first thing that I did) - and jumps are still there:


https://github.com/user-attachments/assets/d23fed95-2591-4a04-8922-60d1574960ed

So it tend me think it's not react-compiler magic that introduces undesired behavior 🪄

**Comment by ikevin127** [comment]
> If anyone can run git bisect to identify the PR that we can revert to fix this, the reward wont miss you.

Sorry, I did not look into the **why**. All I did was to bisect as @mountiny asked on Slack, in order find the offending commit / PR, which could hopefully be used to fix the issue, respectively the deploy block.

##

On a quick look it doesn't make much sense why the PRs changes would cause **this** issue - all I was able to confirm with the bisect is that **it does**.

I'll leave it to the brilliant minds to figure out **why** 🤔

**Comment by kavimuru** [comment]
@Julesssss Issue fixed in iOS

https://github.com/user-attachments/assets/9ff3be5f-17a1-41cc-9162-20c96db4a85f

**Comment by Julesssss** [comment]
The final check will be to verify the Android keyboard issue. As this workflow is currently broken on main I am generating a manual workflow build [here](https://github.com/Expensify/App/actions/runs/9940745955).

**Comment by Julesssss** [comment]
Verified the fix [here](https://github.com/Expensify/App/pull/45403#issuecomment-2228649823) on Android. Checking it off 🤩

**Comment by mountiny** [comment]
$500 to @allgandalf for their help with testing and reviewing

$125 to @ikevin127 for their help identifying the offending commit, @kirillzyusko do you think this means that the react-compiler has somehow changed the memoization of that file or what could be the RCA? @ikevin127 mentioned that for them the problem was resolved reverting before that commit

**Comment by kirillzyusko** [comment]
> do you think this means that the react-compiler has somehow changed the memoization of that file or what could be the RCA?

Well, I don't think so, because I tried to disable `react-compiler` at all and the issue was still there 🤷‍♂️

**Comment by garrettmknight** [comment]
Thanks @Julesssss !

**Comment by ikevin127** [comment]
Payments summary: https://github.com/Expensify/App/issues/45152#issuecomment-2229535918.
